File: C#/0999. Available Captures for Rook.cs
public class Solution {
    public int NumRookCaptures(char[][] board) {
        int rowA=0;
        int columnA=0;
        bool flag=false;
        for(int r=0;r<board.Length;r++){
            for (int c=0;c<board[0].Length;c++){
                if (board[r][c]=='R'){
                    rowA=r;
                    columnA=c;
                    flag=true;
                    break;
                }
            }
            if (flag==true){break;}
        }
        int result=0;
        
        int row=rowA-1;
        int column=columnA;
        while(row>=0){
            if(board[row][column]=='B'){
                break;
            }
            else if(board[row][column]=='p'){
                result+=1;
                break;
            }
            else{
                row-=1;
            }
        }
        
        row=rowA;
        column=columnA-1;
        while(column>=0){
            if(board[row][column]=='B'){
                break;
            }
            else if(board[row][column]=='p'){
                result+=1;
                break;
            }
            else{
                column-=1;
            }
        }
        
        row=rowA+1;
        column=columnA;
        while(row<board.Length){
            if(board[row][column]=='B'){
                break;
            }
            else if(board[row][column]=='p'){
                result+=1;
                break;
            }
            else{
                row+=1;
            }
        }
        
        row=rowA;
        column=columnA+1;
        while(column<board[0].Length){
            if(board[row][column]=='B'){
                break;
            }
            else if(board[row][column]=='p'){
                result+=1;
                break;
            }
            else{
                column+=1;
            }
        }
        
        return result;
    }
}

File: archives/contest/0999. Available Captures for Rook.py
class Solution:
    def numRookCaptures(self, board: List[List[str]]) -> int:
        for row in range(len(board)):
            for column in range(len(board[0])):
                if board[row][column]=='R':
                    return self.subnumRookCaptures(row,column,board)
    
    def subnumRookCaptures(self,row,column,board):
        rep=0
        
        i,j=row,column
        while row-1>=0 and board[row-1][column]=='.':
            row-=1
        if row-1>=0 and board[row-1][column]=='p':
            rep+=1
            
        row,column=i,j
        while column-1>=0 and board[row][column-1]=='.':
            column-=1
        if column-1>=0 and board[row][column-1]=='p':
            rep+=1
        
        row,column=i,j
        while row+1<len(board) and board[row+1][column]=='.':
            row+=1
        if row+1<len(board) and board[row+1][column]=='p':
            rep+=1
        
        row,column=i,j
        while column+1<len(board[0]) and board[row][column+1]=='.':
            column+=1
        if column+1<len(board[0]) and board[row][column+1]=='p':
            rep+=1
        
        return rep
            
